src/states/common/approved.rs

```diff
@@ -413,13 +415,26 @@ pub trait Approved: Base {
             self, their_pub_id
         );
 
-        self.peer_map_mut()
-            .insert(their_pub_id, their_conn_info.clone());
+        self.peer_map_mut().connect(their_conn_info.clone());
         self.send_direct_message(&their_conn_info, DirectMessage::ConnectionResponse);
 
         Ok(())
     }
 
+    fn disconnect_by_id_lookup(&mut self, pub_id: &PublicId) {
+        if let Some(node) = self.chain().get_p2p_node(pub_id.name()) {
+            let node = node.clone();
+            self.disconnect(&node.peer_addr());
```

Comment:
clone or copy the peer_addr instead of the full node since that is what you use.